cmake/modules/AddSwift.cmake

```diff
@@ -1164,6 +1164,19 @@ function(_add_swift_library_single target name)
     list(APPEND link_flags "-fuse-ld=gold")
   endif()
 
+  CMAKE_MINIMUM_REQUIRED(VERSION 2.8 FATAL_ERROR)
+  INCLUDE(CheckCXXCompilerFlag)
+  SET(CMAKE_REQUIRED_FLAGS "-Wl,-z,nocopyreloc")
+  CHECK_CXX_COMPILER_FLAG("" LINKER_SUPPORTS_NOCOPYRELOC)
+  IF(LINKER_SUPPORTS_NOCOPYRELOC)
+      list(APPEND link_flags ${CMAKE_REQUIRED_FLAGS})
+  ENDIF()
+  SET(CMAKE_REQUIRED_FLAGS "-Wl,-z,noextern-protected-data")
+  CHECK_CXX_COMPILER_FLAG("" LINKER_SUPPORTS_NOEXTERN_PROTECTED_DATA)
+  IF(LINKER_SUPPORTS_NOEXTERN_PROTECTED_DATA)
+      list(APPEND link_flags ${CMAKE_REQUIRED_FLAGS})
+  ENDIF()
```

Comment:
I don't think you should `include()` or run tests in this function.  This function is executed many times.

You should run the test once in the top-level CMakeLists.txt.

Also, our CMake code uses lowercase for function names, and two spaces for indentation, could you make it consistent?